crates/topos-p2p/src/behaviour/gossip.rs

```diff
@@ -158,49 +156,42 @@ impl NetworkBehaviour for Behaviour {
         if self.tick.poll_tick(cx).is_ready() {
             // Publish batch
             if !self.echo_queue.is_empty() {
-                let mut echos = Batch { data: Vec::new() };
+                let mut echos = Batch {
+                    messages: Vec::new(),
+                };
                 for _ in 0..self.batch_size {
-                    if let Some(data) = self.echo_queue.pop_front() {
-                        echos.data.push(data);
+                    if let Some(request) = self.echo_queue.pop_front() {
+                        echos.messages.push(request);
                     } else {
                         break;
                     }
```

Comment:
Would this work here?

```rust
let echoes = Batch{messages: self.echo_queue.drain(0..self.batch_size).collect()};
```

Reply:
So this didn't work in the end?

Reply:
https://github.com/topos-protocol/topos/pull/383/commits/e6a292ace6fea11b90d865ee1d2f28c4dd033eb3